keanu-project/src/test/java/io/improbable/keanu/algorithms/SampleStatsTest.java

```diff
@@ -0,0 +1,53 @@
+package io.improbable.keanu.algorithms;
+
+import io.improbable.keanu.distributions.ContinuousDistribution;
+import io.improbable.keanu.distributions.continuous.Uniform;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+public class SampleStatsTest {
+    @Test
+    public void autocorrAtLagZeroIsOne() {
+        double[] samples = generateUniformRandomArray(20);
+        double[] autocorr = SampleStats.autocorrelation(samples);
+        assertEquals(autocorr[0], 1.0, 0.001);
```

Comment:
nit: delta can be a lot smaller (e.g. `1e-10`)?

Reply:
Good point. This can be reduced to 0.0 as it should be exactly equal. Have done this.